lib/fs/modules/LinuxMount.rb

```diff
@@ -309,23 +309,27 @@ def expandLinks(p)
     #
     components.each do |c|
       ncp = File.join(cp, c)
-      #
-      # Each file system know how to check for,
-      # and read its own links.
-      #
-      fs, lp = getFsPathBase(ncp)
-      if fs.fileSymLink?(lp)
-        sl = getSymLink(fs, lp)
-        if sl[0, 1] == '/'
-          cp = sl
-        else
-          cp = File.join(cp, sl)
-        end
+      cp = follow_all_symlinks(ncp)
+    end
+    (cp)
+  end
+
+  def follow_all_symlinks(link_ptr)
+    #
+    # Each filesystem knows how to check for,
+    # and read its own links.
+    #
+    no_more_links = nil
+    until no_more_links
+      filesys, link_ptr = getFsPathBase(link_ptr)
+      if filesys.fileSymLink?(link_ptr)
+        symlink = getSymLink(filesys, linkptr)
+        link_ptr = symlink[0, 1] == '/' ? symlink : File.join(comp, symlink)
```

Comment:
It appears `comp` isn't set.
I'm not sure what the value of `comp` should be when traversing nested, relative symlinks. That may require some investigation.

Reply:
Oh frig.  I copied the actual code into a different copy of the file from what I tested.  Hold on let me recommit this with the *correct* code!